### ISSUE TITLE
[19.03] python3.pkgs.joblib fails to build due to test failure

### ISSUE THREAD

**Issue by CMCDragonkai** [NixOS/nixpkgs#60248]
Joblib when attempting to build on 19.03 `import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/cf3e277dd0bd710af0df667e9364f4bd80c72713.tar.gz) {}` fails:

```
=================================== FAILURES ===================================
________________________________ test_disk_used ________________________________

tmpdir = local('/build/pytest-of-nixbld/pytest-0/test_disk_used0')

    def test_disk_used(tmpdir):
        cachedir = tmpdir.strpath
        # Not write a file that is 1M big in this directory, and check the
        # size. The reason we use such a big file is that it makes us robust
        # to errors due to block allocation.
        a = array.array('i')
        sizeof_i = a.itemsize
        target_size = 1024
        n = int(target_size * 1024 / sizeof_i)
        a = array.array('i', n * (1,))
        with open(os.path.join(cachedir, 'test'), 'wb') as output:
            a.tofile(output)
>       assert disk_used(cachedir) >= target_size
E       AssertionError: assert 1 >= 1024
E        +  where 1 = disk_used('/build/pytest-of-nixbld/pytest-0/test_disk_used0')

joblib/test/test_disk.py:32: AssertionError
============= 1 failed, 1031 passed, 79 skipped in 124.63 seconds ==============
[INFO:MainProcess:MainThread] process shutting down
[DEBUG:MainProcess:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:MainProcess:MainThread] Interpreter shutting down. Waking up queue_manager_threads [(<Thread(QueueManagerThread, started daemon 140737211524864)>, <joblib.externals.loky.process_executor._ThreadWakeup object at 0x7ffff4c28048>)]
[DEBUG:MainProcess:QueueManagerThread] queue management thread shutting down
[DEBUG:MainProcess:QueueManagerThread] closing call_queue
[DEBUG:MainProcess:QueueManagerThread] telling queue thread to quit
[DEBUG:MainProcess:QueueManagerThread] joining processes
[DEBUG:MainProcess:QueueFeederThread] feeder thread got sentinel -- exiting
[DEBUG:MainProcess:QueueManagerThread] queue management thread clean shutdown of worker processes: []
[DEBUG:MainProcess:MainThread] running the remaining "atexit" finalizers
builder for '/nix/store/sk913x41632c4413r87qpcddz9fgw43i-python3.6-joblib-0.13.2.drv' failed with exit code 1
```

I saw this error before here: 
https://github.com/NixOS/nixpkgs/issues/54875#issuecomment-458787619

**Comment by dotlambda** [comment]
> builder for '/nix/store/sk913x41632c4413r87qpcddz9fgw43i-python3.6-joblib-0.13.2.drv' failed with exit code 1

That's definitely not `python3.pkgs.joblib`. `python36.pkgs.joblib` also builds fine. Please provide info on how to reproduce this.

**Comment by CMCDragonkai** [comment]
Just using nix repl with the pkg set brought in by `import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/cf3e277dd0bd710af0df667e9364f4bd80c72713.tar.gz) {}`. And then `:b pkgs.python36Packages.joblib` or `:b pkgs.python37Packages.joblib`.

**Comment by FlorianFranzen** [comment]
@dotlambda I also reported and fixed these test failures in PR #57176 (never merged), so I would consider them reproducible, at least when running NixOS on BTRFS (or a LUKS volume).

@CMCDragonkai What file system are you using? Do you encrypt your hard drive or have file system level compression enabled?

**Comment by dotlambda** [comment]
@FlorianFranzen I merged your fix as 604dae6797f4e2d83a263c3f4ffb639dce3bce04. Thanks a lot!

**Comment by CMCDragonkai** [comment]
I use ZFS. What's the source problem.
